frontend/src/components/dashboard/aliases/AliasDeletionButton.test.tsx

```diff
@@ -0,0 +1,140 @@
+import { render, screen, within } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import { mockConfigModule } from "../../../../__mocks__/configMock";
+import { mockLocalizedModule } from "../../../../__mocks__/components/Localized";
+import { getMockRandomAlias } from "../../../../__mocks__/hooks/api/aliases";
+import { mockUseL10nModule } from "../../../../__mocks__/hooks/l10n";
+
+import { AliasDeletionButton } from "./AliasDeletionButton";
+
+jest.mock("../../../config.ts", () => mockConfigModule);
+jest.mock("../../../hooks/gaViewPing.ts");
+jest.mock("../../../hooks/l10n.ts", () => mockUseL10nModule);
+jest.mock("../../../components/Localized.tsx", () => mockLocalizedModule);
+
+describe("<AliasDeletionButton>", () => {
+  it("displays a usable button to delete an alias", () => {
+    render(
+      <AliasDeletionButton alias={getMockRandomAlias()} onDelete={jest.fn()} />
+    );
+
+    const button = screen.getByRole("button", {
+      name: "l10n string: [profile-label-delete], with vars: {}",
+      hidden: false,
+    });
+
+    expect(button).toBeInTheDocument();
+  });
+
+  it("displays a confirmation prompt, with an unchecked checkbox and a disabled button", async () => {
+    render(
+      <AliasDeletionButton alias={getMockRandomAlias()} onDelete={jest.fn()} />
+    );
+
+    const button = screen.getByRole("button", {
+      name: "l10n string: [profile-label-delete], with vars: {}",
+      hidden: false,
+    });
+
+    await userEvent.click(button);
+
+    const prompt = screen.getByRole("dialog");
+    const promptCheckbox = within(prompt).getByRole("checkbox");
+    const promptButton = within(prompt).getByRole("button", {
+      name: "l10n string: [profile-label-delete], with vars: {}",
+    });
+
+    expect(promptCheckbox).not.toBeChecked();
+    expect(promptButton).toBeDisabled();
+  });
+
+  it("enables the delete button on the confirmation prompt, once the checkbox is checked", async () => {
+    render(
+      <AliasDeletionButton alias={getMockRandomAlias()} onDelete={jest.fn()} />
+    );
+
+    const button = screen.getByRole("button", {
+      name: "l10n string: [profile-label-delete], with vars: {}",
+    });
+
+    await userEvent.click(button);
+
+    const prompt = screen.getByRole("dialog");
+    const promptCheckbox = screen.getByRole("checkbox");
+    const promptButton = within(prompt).getByRole("button", {
+      name: "l10n string: [profile-label-delete], with vars: {}",
```

Comment:
Ideally, selectors find elements the same way users do, e.g. by looking for a button if possible, or label text if not. Class names are not visible to users and therefore more likely to change. See also https://testing-library.com/docs/queries/about/#priority

That idea is somewhat less powerful due to localisation abstracting over the actual text that the user sees, but it still makes sure that we only need to update the test if we update how the user interacts with the website, rather than if we just move around the HTML structure.